pkgs/development/python-modules/crochet/default.nix

```
@@ -0,0 +1,45 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  twisted,
+  wrapt,
+}:
+
+buildPythonPackage rec {
+  pname = "crochet";
+  version = "2.1.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "itamarst";
+    repo = "crochet";
+    tag = version;
+    hash = "sha256-grymhvCC9zDBKhNnQC0o07hdLPV5KMWb6HSz/ntSbq8=";
+  };
+
+  # fix for python>=3.12
+  postPatch = ''
+    substituteInPlace versioneer.py \
+      --replace-fail "SafeConfigParser()" "ConfigParser()" \
+      --replace-fail "parser.readfp" "parser.read_file"
+  '';
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    twisted
+    wrapt
+  ];
+
+  pythonImportsCheck = [ "crochet" ];
+
```

```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```

I noticed crochet has pytest tests.

---

Good catch,  thank you.